### PR TITLE
fix memory leak by putting setInterval() in a useeffect

### LINKED ISSUE #80: High memory usage
I left a tab open in Firefox overnight in the background with the wall display open. When I went to the tab this morning it was slow and unresponsive. When I closed the tab, Firefox memory usage declined by 650MB.

### PR DESCRIPTION
**Pull request by rerpha**
Closes #80 

this also fixes a bug that we witnessed a few weeks ago where POLREF got stuck in RESUMING on the wall display.

### CODE REVIEW

File: app/wall/page.tsx [hunk 1]
```diff
@@ -132,40 +130,50 @@ export default function WallDisplay() {
...
+            );
+            if (foundInstrument) {
+              foundInstrument.runstatePV = instPrefix + runstatePV;
+              // Subscribe to the instrument's runstate PV
+              sendJsonMessage({
+                type: PVWSRequestType.subscribe,
+                pvs: [foundInstrument.runstatePV],
+              });
+            }
+          });
```
Comment (Tom-Willemsen) [inline app/wall/page.tsx:152]: Are you sure this is a correct usage of `.map`?

We're not binding anything to the result of this call, if that's correct and we really want to be mutating the original then `forEach` maybe clearer. `.map` is usually recommended to explicitly avoid mutating the originals though?

Comment (rerpha) [reply app/wall/page.tsx:152]: yeah a forEach is much clearer and we've already copied with the spread above (`[...prev]`) 
will implement.

Comment (Tom-Willemsen) [reply app/wall/page.tsx:152]: Thanks, there were a couple of other places like this, only commented on one

**Review by Tom-Willemsen — APPROVED** [review]
This refactor got a bit big!

Functionally, did leave a tab open on wall display overnight and didn't seem to use significantly more memory after a long time. Refactors look good although I didn't go through all with a fine-tooth comb...

**Comment by rerpha** [comment]
it did, sorry, i got a bit carried away as I noticed a few issues with the wall display not updating sometimes etc. 
let's merge for now if you're happy. I think there's a lot more tests that can/should be added for some of the components which don't have snapshot tests, we should have most things typed now so that should help a bit.

**Pull request merged**